rust/src/database/verkle/variants/simple/node.rs

```diff
@@ -0,0 +1,581 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+#[cfg(test)]
+use crate::{database::verkle::compute_commitment::compute_leaf_node_commitment, types::Key};
+use crate::{database::verkle::crypto::Commitment, types::Value};
+
+/// A node in the simple in-memory Verkle trie.
+#[cfg_attr(not(test), expect(unused))]
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug)]
+pub enum Node {
+    Empty,
+    Inner(InnerNode),
+    Leaf(LeafNode),
+}
+
+#[cfg(test)]
+impl Node {
+    /// Returns the value associated with the given key, or the default value if
+    /// the key does not exist.
+    pub fn get(&self, key: &Key, depth: u8) -> Value {
+        match self {
+            Node::Empty => Value::default(),
+            Node::Inner(inner) => inner.get(key, depth),
+            Node::Leaf(leaf) => leaf.get(key),
+        }
+    }
+
+    /// Sets the value for the given key.
+    /// Consumes the node and returns an updated version.
+    pub fn set(self, key: &Key, depth: u8, value: &Value) -> Node {
+        match self {
+            Node::Empty => {
+                // While conceptually it would suffice to create a leaf node here,
+                // Geth always creates an inner node (and we want to stay compatible).
+                let inner = InnerNode::new();
+                inner.set(key, depth, value)
```

Comment:
```suggestion
                if depth == 0 {
                    // While conceptually it would suffice to create a leaf node here,
                    // Geth always creates an inner node (and we want to stay compatible).
                    let inner = InnerNode::new();
                    inner.set(key, depth, value)
                } else {
                    let leaf = LeafNode::new(key);
                    leaf.set(key, depth, value)
                }
```

Reply:
I like it!